app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -438,6 +438,10 @@ public function addAttributeToSelect($attribute, $joinType = false)
             } else {
                 $attrInstance = Mage::getSingleton('eav/config')
                     ->getAttribute($this->getEntity()->getType(), $attribute);
+                // failure to resolve an attribute from the eav config implies it does not exist
+                if (empty($attrInstance)) {
+                    return $this;
```

Comment:
Shouldnt we log situations like this? Typo in the attribute name is a  bug, a deweloper would like to know about

Reply:
It's not necessarily a bug, it could just be a deleted attribute that was referenced somewhere. But perhaps throw an exception in developer mode and log it in production mode. Let me know what you prefer and I defer to you and make the change.

Reply:
For me, nothing, or only a log when dev mode is enabled.

Reply:
I also don't care, if an attribute disappears there will be an errors in some template or some other code without the need to do anything here, although maybe it would be a bit cleaner